include/kwk/detail/sequence/prefilled.hpp

```diff
@@ -269,44 +269,29 @@ namespace kwk::__
     }
 
     // Dynamic access - We rely on KUMI internals for speed and ease of detection
-    KWK_TRIVIAL constexpr auto operator[](std::convertible_to<std::size_t> auto i) const noexcept
-    requires(is_dynamic && static_size>0)
+    KWK_TRIVIAL constexpr auto& operator[](std::convertible_to<std::uint32_t> auto i) noexcept
+    requires(is_fully_dynamic && static_size>0)
     {
-      static_assert(is_homogeneous, "[KWK] - Dynamic axis access requires homogeneous extent types.");
-      auto const idx = static_cast<std::size_t>(i);
-
-      KIWAKU_ASSERT ( idx<static_size
-                    , "[KWK] - Out of bounds access at index: " << idx << " for a max. of " << static_size
-                    );
+      auto const idx = static_cast<std::uint32_t>(i);
 
       KIWAKU_ASSERT ( contains(idx)
                     ,   "[KWK] - Access at index "  << idx
                     <<  " overwrites a compile-time value of " << kumi::to_tuple(*this)
                     );
 
-      // KUMI internal shortcut
-      if constexpr(dynamic_size==1) return storage().impl.member0;
-      else                          return storage().impl.members[idx];
+      return get_indexed( idx );
     }
 
-    KWK_TRIVIAL constexpr auto& operator[](std::convertible_to<std::size_t> auto i) noexcept
-    requires(is_dynamic && static_size>0)
+    KWK_TRIVIAL constexpr auto operator[](std::convertible_to<std::uint32_t> auto i) const noexcept
+    requires(is_fully_dynamic && static_size>0)
     {
-      static_assert(is_homogeneous, "[KWK] - Dynamic axis access requires homogeneous extent types.");
-      auto const idx = static_cast<std::size_t>(i);
-
-      KIWAKU_ASSERT ( idx<static_size
-                    , "[KWK] - Out of bounds access at index: " << idx << " for a max. of " << static_size
-                    );
-
-      KIWAKU_ASSERT ( contains(idx)
-                    ,   "[KWK] - Access at index "  << idx
-                    <<  " overwrites a compile-time value of " << kumi::to_tuple(*this)
-                    );
+      return const_cast<prefilled &>(*this).get_indexed(static_cast<std::uint32_t>(i));
+    }
 
-      // KUMI internal shortcut
-      if constexpr(dynamic_size==1) return storage().impl.member0;
-      else                          return storage().impl.members[idx];
+    KWK_TRIVIAL constexpr auto operator[](std::convertible_to<std::uint32_t> auto i) const noexcept
+    requires(!(is_fully_dynamic && static_size>0))
+    {
+      return as_array()[i]; // rely on std::array bounds checking
```

Comment:
well as_array was suboptimal. I thought this was geenrting better code.

Reply:
i think the main issue it that the way it was implemented actually did not work (compile) so this was a quick-fix...we'll have to go through all this live i guess

Reply:
in which scenario did it din't work ?

Reply:
i think when you used it on a shape that is only is_dynamic but not fully_dynamic (as the original constraint specified): you would index into a kumi tuple internal array that was shorter in length then the shape you were 'randomly' accessing